pkg/systemd/page-status.jsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+import React from "react";
+import { page_status } from "notifications";
+
+const _ = cockpit.gettext;
+
+function fragfmt(fmt, arg) {
+    var index = fmt.indexOf("$0");
+    if (index >= 0)
+        return <>{fmt.slice(0, index)}{arg}{fmt.slice(index + 2)}</>;
+    else
+        return fmt;
+}
+
+function fragjoin(arr, sep) {
+    function step(i) {
+        if (arr.length == 0)
+            return null;
+        if (arr.length == i + 1)
+            return arr[i];
+        return <>{arr[i]}{sep}{step(i + 1)}</>;
+    }
+    return step(0);
+}
+
+export class PageStatusNotifications extends React.Component {
+    constructor() {
+        super();
+        this.state = { };
+        this.on_page_status_changed = () => { this.setState({ }) };
```

Comment:
Seems a bit weird to define this in the constructor instead of just as a regular method like the others, but it's clear enough. (Also, you don't need these extra `{ } ;` with arrow functions -- they really should be simple expressions in most cases)

Reply:
`on_page_status_changed` needs to be defined with arrow syntax so that `this` has the right value in its body.

When we define it as a member of the class and the use it in `addEventListener("changed", on_page_status_changed)`, `this` will be wrong when it is eventually called.

Reply:
I like to use braces when the body is more like a statement than an expression, and we care more about its side effects than its value.  But I'll remove the { }.

`x => 2*x` but `x => { list.push(x) }`

Reply:
Ah, so that constructor assignment is the way to avoid these `.bind(this)` calls? Nice!